javascript/code-challenges/challenges-07.test.js

```diff
@@ -0,0 +1,383 @@
+
+For example, if the input is 'Welcome', the output will be:
+['Welcome', 'elcome', 'lcome', 'come', 'ome', 'me', 'e', ''].
+------------------------------------------------------------------------------------------------ */
+
+const howMuchPencil = (str) => {
+  let result = [];
+  
+  for (let i = 0; i <= str.length; i+=1) {
+    result.push(str.slice(i));
+    
+  }
+  return result;
+  
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function name wordsToCharList that, given a string as input, returns a new array where every element is a character of the input string.
+
+For example, wordsToCharList('gregor') returns ['g','r','e','g','o','r'].
+------------------------------------------------------------------------------------------------ */
+
+const wordsToCharList = (arr) => {
+  let returnArray = arr.split("");
+  return returnArray;
+};
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 6
+
+You are making a grocery list for ingredients needed in the gruffalo crumble recipe, below. Rather than taking the entire recipe, you only want a list of the item names.
+
+Write a function named listFoods that takes in the recipe and returns a new array of the food items without any amount or units. Just the name. For example, '1 cup flour' will return 'flour'.
+
+Use slice for this function, maybe more than once. The Array.indexOf() method may also be helpful.
+
+Do not use split for this function.
+------------------------------------------------------------------------------------------------ */
+
+const gruffaloCrumble = {
+  name: 'How to make a Gruffalo Crumble',
+  ingredients: [
+    '1 medium-sized Gruffalo',
+    '8 pounds oats',
+    '2 pounds brown sugar',
+    '4 pounds flour',
+    '2 gallons pure maple syrup',
+    '16 cups chopped nuts',
+    '1 pound baking soda',
+    '1 pound baking powder',
+    '1 pound cinnamon',
+    '6 gallons melted butter',
+    '2 gallons fresh water',
+  ],
+  steps: [
+    'Pre-heat a large oven to 375',
+    'De-prickle the gruffalo',
+    'Sprinkle with cinnamon, sugar, flour, and nuts',
+    'Mix until evenly distributed',
+    'Grease a 3-foot x 3-foot casserole dish',
+    'Combine gruffalo compote with water to maintain moisture in the oven',
+    'Fold together remaining ingredients to make the crisp',
+    'Spread the crisp evenly over the gruffalo mixture',
+    'Bake for 12-15 hours',
+  ]
+};
+
+
+const listFoods = (recipe) => {
+  let result = [];
+  
+  console.log(result)
+  let word1 = result[0]
+  
+  // word1.split(' ');
+  console.log(word1);
+
+  for (let i = 0; i < recipe.ingredients.length; i+=1) {
+    result.push(recipe.ingredients);
+    // const items = result[i];
+    
+    // words.split(' ');
+    // console.log(words[3]);
+  }
+  
+  return result;
+};
```

Comment:
Good work on the challenges! You are really close on this one :)

- We can use map(), forEach() on this one (if you use forEach then line 138 can be kept)
- You are correct on iterating through `recipe.ingredients` and for each string, we need to find the index of the first space in the string (we could use `indexOf()` (this is saved to a variable for later use).
- Once the first space is found, we need to remove everything before that space and the space itself (we can use `slice()` for this)
- Next, we need to track the NEXT index of the second location of the next empty space in string (this might involve updating value of variable that holds the index of first space in string).
- We basically keep slicing away at this string until there are no spaces. Once that is done, if using forEach or iterating with for loop, then we can push our result into our array.

I hope this helps a bit 👍